### DISCUSSION TITLE
How to configure PP to write sidecar files beside originals

### DISCUSSION THREAD

**Discussion by hairlesshobo** [photoprism/photoprism#2056] [Q&A] [answered]
I have searched through the docs, GH issues and GH discussions and have had no luck finding an answer to this, or whether it is even possible. I am looking for a config option to have sidecar files (both `yml` metadata and `jpg` RAW conversion files) that are created by PhotoPrism to be written alongside the files in the originals folder. At the moment, they are written to the sidecar storage directory.

When reading the FAQ in the docs "[what are sidecar files and where do i find them?](https://docs.photoprism.app/getting-started/faq/#what-are-sidecar-files-and-where-do-i-find-them)" it is stated that "New sidecar files are created in the storage folder by default, so the originals folder can be mounted read-only." Since it states that it does this /by default/, does that mean the behavior can be changed so that these sidecar files are written back to the originals folder?

Thanks in advance.

**Comment by lastzero** [comment, marked as answer]
`PHOTOPRISM_SIDECAR_PATH: "."` should work!?

**Reply by hairlesshobo** [reply]
Wow, I feel silly for missing that simple solution. Thanks for the incredibly quick reply!

**Reply by al13nus** [reply]
What will happen, if I write sidecar files beside originals and re-index the library? Total mess?

**Reply by al13nus** [reply]
Background: I have bad gut feeling groving my /opt-Folder, where all the docker container are saving their settings, because transcoded files a not really a perfect match to be saved there BUT saving beside originals sounds also wrong. At the end its compatible copy of the original. Looks like having a second folder something like /transcodes beside /original is my favorit.

**Reply by al13nus** [reply]
@lastzero suggested to put the sidecars in hidden folder under originals: `.photoprism`. Would it look like` PHOTOPRISM_SIDECAR_PATH: ".photoprism"` then?

**Reply by al13nus** [reply]
Ok, this creates in every `YEAR/MONTH`-Folder a new `.photoprism`-Folder

**Reply by al13nus** [reply]
This worked for me: `/photoprism/originals/.photoprism-sidecar`

**Reply by melyux** [reply]
Is there a way to only have the metadata sidecar (e.g. yml) next to the originals while keeping the converted sidecars (e.g. JPEGs for original HEIF images) separate? The reason is that metadata sidecar files are important and irreplaceable, but the converted sidecars can always be regenerated and are throwaway files.

Say, I edit the location of a photo through PhotoPrism. It's written into a sidecar yaml file. That's original data and irreplaceable.
Say, PhotoPrism converts my iPhone photos to a web-friendly JPEG format. It's written into a sidecar JPEG file. That can just be regenerated by PhotoPrism anytime it wants, and so replaceable.

**Reply by lastzero** [reply]
It is not possible to configure a path by type right now.. The sidecar YAML files include the same information as a SQL dump of the database, so they are not completely irreplaceable either.

**Reply by melyux** [reply]
@lastzero If the sidecar yaml files are deleted, does PhotoPrism regenerate them from the database?

**Reply by lastzero** [reply]
Yes

**Reply by Johnny5AtGithub** [reply]
I'm currently on PhotoPrism® CE Build 250707-d28b3101e. Adding the file `.photoprism-sidecar` worked for me. 
Setting `PHOTOPRISM_SIDECAR_PATH` configuration on docker DID NOT work for me.

I have made sure to have restarted the entire docker between each attempt to make sure any setting has been read.
